models/gif_cond_exp.h

```diff
@@ -0,0 +1,450 @@
+
+    /** External DC current. */
+    double_t I_e_;
+
+    Parameters_(); //!< Sets default parameter values
+
+    void get( DictionaryDatum& ) const; //!< Store current values in dictionary
+    void set( const DictionaryDatum& ); //!< Set values from dictionary
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * State variables of the model.
+   */
+  struct State_
+  {
+    //! Symbolic indices to the elements of the state vector y
+    enum StateVecElems
+    {
+      V_M = 0,
+      G_EXC,
+      G_INH,
+      STATE_VEC_SIZE
+    };
+
+    double_t
+      y_[ STATE_VEC_SIZE ]; //!< neuron state, must be C-array for GSL solver
+
+    double_t y0_;  //!< This is piecewise constant external current
+    double_t sfa_; //!< This is the change of the 'threshold' due to adaptation.
+    double_t stc_; //!< Spike-triggered current.
+
+    std::vector< double_t > sfa_elems_; //!< Vector of adaptation parameters.
+    std::vector< double_t >
+      stc_elems_; //!< Vector of spike-triggered parameters.
+
+    double_t i_syn_ex_; //!< postsynaptic current for exc.
+    double_t i_syn_in_; //!< postsynaptic current for inh.
+
+    int_t r_ref_; //!< absolute refractory counter (no membrane potential
+    // propagation)
+
+    State_( const Parameters_& ); //!< Default initialization
+    State_( const State_& );
+    State_& operator=( const State_& );
+
+    void get( DictionaryDatum&, const Parameters_& ) const;
+    void set( const DictionaryDatum&, const Parameters_& );
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Buffers of the model.
+   */
+  struct Buffers_
+  {
+    Buffers_( gif_cond_exp& );
+    Buffers_( const Buffers_&, gif_cond_exp& );
+
+    /** buffers and sums up incoming spikes/currents */
+    RingBuffer spike_exc_;
+    RingBuffer spike_inh_;
+    RingBuffer currents_;
+
+    //! Logger for all analog data
+    UniversalDataLogger< gif_cond_exp > logger_;
+
+    /** GSL ODE stuff */
+    gsl_odeiv_step* s_;    //!< stepping function
+    gsl_odeiv_control* c_; //!< adaptive stepsize control function
+    gsl_odeiv_evolve* e_;  //!< evolution function
+    gsl_odeiv_system sys_; //!< struct describing system
+
+    // IntergrationStep_ should be reset with the neuron on ResetNetwork,
+    // but remain unchanged during calibration. Since it is initialized with
+    // step_, and the resolution cannot change after nodes have been created,
+    // it is safe to place both here.
+    double_t step_;          //!< step size in ms
+    double IntegrationStep_; //!< current integration time step, updated by GSL
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Internal variables of the model.
+   */
+  struct Variables_
+  {
+    std::vector< double_t > P_sfa_;
+    std::vector< double_t > P_stc_;
+
+    librandom::RngPtr rng_; // random number generator of my own thread
+
+    int_t RefractoryCounts_;
```

Comment:
Same as for ``r_ref_``, could you change this to ``unsigned int refractory_counts_;``?